### PR TITLE
Update concepts, conventions, events, tileentities to 1.13

### PR DESCRIPTION
**Pull request by williewillus**

### CODE REVIEW

File: docs/concepts/internationalization.md [hunk 1]
```diff
@@ -3,51 +3,39 @@ Internationalization and localization
...
+Block, Item and a few other Minecraft classes have built-in translation keys used to display their names. These translation keys are specified by overriding `getTranslationKey()`. Item also has `getTranslationKey(ItemStack)` which can be overridden to provide different translation keys depending on ItemStack NBT.
 
-By default, `getTranslationKey()` will return `tile.` or `item.` prepended to whatever was set with `setTranslationKey(String)`, and ItemBlock will inherit the unlocalized name of its Block. Additionally, `.name` is always appended to the translation key before it is localized. For example `item.setTranslationKey("examplemod.example_item")` effectively requires the following line in a language file:
+By default, `getTranslationKey()` will return `block.` or `item.` prepended to the registry name of the block or item, with the colon replaced by a dot. ItemBlocks will take their corresponding Block's translation key by default. For example, an item with ID `examplemode:example_item` effectively requires the following line in a language file:
```
Comment (LukeGi) [inline docs/concepts/internationalization.md]: probably meant to be example**mod** not example**mode**

**Comment by williewillus** [comment]
anything else?

File: docs/conventions/loadstages.md [hunk 2]
```diff
@@ -1,50 +1,51 @@
...
+  } 
+  
+  private void commonSetup(FMLCommonSetupEvent evt) { ... }
+}
+```
 
 !!! warning
+    All four of the below events are called for all mods in parallel. That is, all mods will concurrently receive common setup, FML will wait for 
+    them all to finish, then all mods will concurrently receive sided setup, and so forth.
+    Mods *must* take care to be thread safe, especially when calling other mods' API's and accessing Vanilla systems, which are not thread safe in general.
```
Comment (tterrag1098) [inline docs/conventions/loadstages.md]: Should add a short mention of, or dummy link to info about, the `DeferredWorkQueue`.

File: docs/conventions/loadstages.md [hunk 3]
```diff
@@ -1,50 +1,51 @@
...
+Here, mods should send messages to all other mods they are interested in integrating with, using the `InterModComms.sendTo()` method.
 
-Post Init is where your mod usually does things which rely upon other mods.
-This stage's event is the `FMLPostInitializationEvent`.
-Common actions to preform in postInit are:
+## IMC Process
 
-  * Mod compatibility, or anything which depends on other mods' init phases being finished.
+Here, mods should process all the messages they have received from other mods and set up integrations appropriately. A mod may retrieve the messages
+that have been sent to it using the `InterModComms.getMessages()` method.
```
Comment (tterrag1098) [inline docs/conventions/loadstages.md]: Random newline?

File: docs/conventions/locations.md [hunk 4]
```diff
@@ -5,21 +5,17 @@ Minecraft expects certain parts of your project to be in certain locations, such
...
 Blockstate definition files are in the JSON format and are in the `./assets/<modid>/blockstates/` folder.
 
 ### Localizations
 
-Localizations are plain-text files with the file extension `.lang` and the name being their [language code](https://msdn.microsoft.com/en-us/library/ee825488(v=cs.20).aspx) such as `en_US`.
-
-!!! note
-
-    If a `pack_format` of 3 is specified in a `pack.mcmeta` file, the language code should be lowercase.
+Localizations are plain-text files with the file extension `.json` and the name being their [language code](https://msdn.microsoft.com/en-us/library/ee825488(v=cs.20).aspx) in lowercase such as `en_us`.
```
Comment (tterrag1098) [inline docs/conventions/locations.md]: Link should go in the index

Comment (tterrag1098) [inline docs/conventions/locations.md]: This needs to link to concepts/i18n as well

File: docs/tileentities/tesr.md [hunk 5]
```diff
@@ -1,18 +1,18 @@
-TileEntitySpecialRenderer
-=========================
+TileEntityRenderer
+==================
 
-A `TileEntitySpecialRenderer` (TESR) is used to render blocks in a way that cannot be represented with a static baked model (JSON, OBJ, B3D, others). A TESR requires the block to have a TileEntity.
+A `TileEntityRenderer` (colloquially called TESR) is used to render blocks in a way that cannot be represented with a static baked model (JSON, OBJ, B3D, others). A TESR requires the block to have a TileEntity.
```
Comment (tterrag1098) [inline docs/tileentities/tesr.md]: Perhaps `previously TileEntitySpecialRenderer, or TESR for short`. We should be moving away from `TESR` as an abbreviation.

File: docs/tileentities/tesr.md [hunk 6]
```diff
@@ -27,9 +27,9 @@ Registering a TESR
...
+A fast TESR can offer performance improvements over a traditional TESR and should be used wherever possible. This is due to the fact that all fast TESR instances are batched together and only issue _one_ combined draw call for all fast TESRs per frame to the GPU. This advantage comes at the cost of making direct OpenGL access via `GlStateManager` or the `GLXX` classes impossible. Instead a fast TESR must only add vertices to the provided `VertexBuffer`, which represents the combined vertex data for all fast TESRs. This allows rendering `IBakedModel`s. An example can be found in Forge's `TileEntityRendererAnimation`.
```
Comment (tterrag1098) [inline docs/tileentities/tesr.md]:
```suggestion
A fast renderer can offer performance improvements over a traditional `TileEntityRenderer` and should be used wherever possible. This is due to the fact that all fast TER instances are batched together and only issue _one_ combined draw call for all fast TERs per frame to the GPU. This advantage comes at the cost of making direct OpenGL access via `GlStateManager` or the `GLXX` classes impossible. Instead a fast renderer must only add vertices to the provided `VertexBuffer`, which represents the combined vertex data for all fast renderers. This allows rendering `IBakedModel`s. An example can be found in Forge's `TileEntityRendererAnimation`.
```
Maybe also define the TER acronym above.

File: docs/tileentities/tileentity.md [hunk 7]
```diff
@@ -12,28 +12,23 @@ More advanced examples exist in mods, such as quarries, sorting machines, pipes,
...
-It is important that your `TileEntity` has a default constructor, so that Minecraft can properly load it.
-After you've created your class, you need to register the `TileEntity`. For this you need to call
-
-```JAVA
-GameRegistry#registerTileEntity(Class<? extends TileEntity> tileEntityClass, ResourceLocation key)
+To register it, listen for the appropriate registry event and create a `TileEntityType`:
+```Java
+@SubscribeEvent
+public static void registerTE(RegistryEvent.Register<TileEntityType> evt) {
+  TileEntityType type = TileEntityType.Builder.create(<factory>).build(null)
```
Comment (tterrag1098) [inline docs/tileentities/tileentity.md]: No rawtypes!

**Comment by williewillus** [comment]
comments addressed

**Pull request merged**